src/LaunchDarkly.ServerSdk.DynamoDB/DynamoDBDataStoreBuilder.cs

```diff
@@ -0,0 +1,159 @@
+    /// <para>
+    /// Obtain an instance of this class by calling <see cref="DynamoDB.DataStore(string)"/>. After calling its methods
+    /// to specify any desired custom settings, wrap it in a <see cref="PersistentDataStoreBuilder"/>
+    /// by calling <see cref="Components.PersistentDataStore(IPersistentDataStoreAsyncFactory)"/>, then pass
+    /// the result into the SDK configuration with <see cref="ConfigurationBuilder.DataStore(IDataStoreFactory)"/>.
+    /// You do not need to call <see cref="CreatePersistentDataStore(LdClientContext)"/> yourself to build
+    /// the actual data store; that will be done by the SDK.
+    /// </para>
+    /// <para>
+    /// The AWS SDK provides many configuration options for a DynamoDB client. This class has
+    /// corresponding methods for some of the most commonly used ones. If you need more sophisticated
+    /// control over the DynamoDB client, you can construct one of your own and pass it in with the
+    /// <see cref="ExistingClient(AmazonDynamoDBClient)"/> method.
+    /// </para>
+    /// <para>
+    /// Builder calls can be chained, for example:
+    /// </para>
+    /// <code>
+    ///     var config = Configuration.Builder("sdk-key")
+    ///         .DataStore(
+    ///             Components.PersistentDataStore(
+    ///                 DynamoDB.DataStore("my-table-name")
+    ///                     .Credentials(myAWSCredentials)
+    ///                     .Prefix("app1")
+    ///                 )
+    ///                 .CacheSeconds(15)
+    ///             )
+    ///         .Build();
+    /// </code>
+    /// </remarks>
+    public sealed class DynamoDBDataStoreBuilder : IPersistentDataStoreAsyncFactory
+    {
+        private AmazonDynamoDBClient _existingClient = null;
+        private AWSCredentials _credentials = null;
+        private AmazonDynamoDBConfig _config = null;
+
+        private readonly string _tableName;
+        private string _prefix = "";
+        
+        internal DynamoDBDataStoreBuilder(string tableName)
+        {
+            _tableName = tableName;
+        }
+
+        /// <summary>
+        /// Specifies an existing, already-configured DynamoDB client instance that the data store
+        /// should use rather than creating one of its own.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// If you specify an existing client, then the other builder methods for configuring DynamoDB
+        /// are ignored.
+        /// </para>
+        /// <para>
+        /// Note that the LaunchDarkly code will <i>not</i> take ownership of the lifecycle of this
+        /// object: in other words, it will not call <c>Dispose()</c> on the <c>AmazonDynamoDBClient</c> when
+        /// you dispose of the SDK client, as it would if it had created the <c>AmazonDynamoDBClient</c> itself.
+        /// It is your responsibility to call <c>Dispose()</c> on the <c>AmazonDynamoDBClient</c> when you are
+        /// done with it.
```

Comment:
The paragraph above describes new behavior. Previously, the DynamoDB integration would dispose of the DynamoDB client on shutdown regardless of whether it was a pre-existing client instance or not. I've changed it so we only dispose of the client if it was created by us, which I believe is more consistent with typical .NET object lifecycle management.